core/domain/prod_validation_jobs_one_off.py

```diff
@@ -84,93 +190,2162 @@ def _fetch_external_models(cls, item):
+            'story_rights_ids': (
+                story_models.StoryRightsModel,
+                [item.id[:item.id.find('-')]]),
+        }
+
+
+class StoryCommitLogEntryModelValidator(BaseCommitLogEntryModelValidator):
+    """Class for validating StoryCommitLogEntryModel."""
+
+    related_model_name = 'story'
+
+    @classmethod
+    def _get_model_id_regex(cls, item):
+        # Valid id: [story/rights]-[story_id]-[story_version].
+        regex_string = '^(story|rights)-%s-\\d*$' % (
+            item.story_id)
+
+        return regex_string
+
+    @classmethod
+    def _get_change_domain_class(cls):
+        return story_domain.StoryChange
+
+    @classmethod
+    def _get_external_id_relationships(cls, item):
+        return {
+            'story_ids': (
+                story_models.StoryModel, [item.story_id]),
+        }
+
+
+class StorySummaryModelValidator(BaseSummaryModelValidator):
+    """Class for validating StorySummaryModel."""
+
+    @classmethod
+    def _get_model_domain_object_instance(cls, item):
+        return story_services.get_story_summary_from_model(item)
+
+    @classmethod
+    def _get_external_id_relationships(cls, item):
+        return {
+            'story_ids': (
+                story_models.StoryModel, [item.id]),
+            'story_rights_ids': (
+                story_models.StoryRightsModel, [item.id]),
+        }
+
+    @classmethod
+    def _validate_node_count(cls, item):
+        """Validate that node_count of model is equal to number of nodes
+        in StoryModel.story_contents.
+
+        Args:
+            item: ndb.Model. StorySummaryModel to validate.
+        """
+        story_model_class_model_id_model_tuples = cls.external_instance_details[
+            'story_ids']
+
+        for (_, _, story_model) in story_model_class_model_id_model_tuples:
+            nodes = story_model.story_contents['nodes']
+            if item.node_count != len(nodes):
+                cls.errors['node count check'].append((
+                    'Entity id %s: Node count: %s does not match the '
+                    'number of nodes in story_contents dict: %s') % (
+                        item.id, item.node_count, len(nodes)))
+
+    @classmethod
+    def _get_related_model_properties(cls):
+        story_model_class_model_id_model_tuples = cls.external_instance_details[
+            'story_ids']
+
+        story_model_properties_dict = {
+            'title': 'title',
+            'language_code': 'language_code',
+            'description': 'description',
+            'story_model_created_on': 'created_on',
+        }
+
+        return [(
+            'story',
+            story_model_class_model_id_model_tuples,
+            story_model_properties_dict
+        )]
+
+    @classmethod
+    def _get_custom_validation_functions(cls):
+        return [cls._validate_node_count]
+
+
+class UserSubscriptionsModelValidator(BaseModelValidator):
+    """Class for validating UserSubscriptionsModels."""
```

Comment:
Consider adding a backend test that iterates through all non-abstract model classes in the codebase and verifies that each one has a corresponding audit one off job in this file.

Reply:
This should be done after I have added jobs for all the models, right? Or should I have a test now as well which just iterates over the all the models in these 9 directories?

Reply:
The former is fine, but please file an issue for this and assign it to yourself. Thanks!

Reply:
Done #6952!